src/BugsnagUnity.mm

```diff
@@ -168,23 +168,45 @@
 
 }
 
-void bugsnag_getEventMetaData(const void *event,const void *instance, const char *tab, void (*callback)(const void *instance, const char *keys[], int keys_size, const char *values[], int values_size)) {
-     NSDictionary* sectionDictionary = [((__bridge BugsnagEvent *)event) getMetadataFromSection:@(tab)];
-     NSArray *keys = [sectionDictionary allKeys];
-     NSArray *values = [sectionDictionary allValues];
-     int count = 0;
-     if ([keys count] <= INT_MAX) {
-       count = (int)[keys count];
-     }
-     const char **c_keys = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
-     const char **c_values = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
-     for (NSUInteger i = 0; i < (NSUInteger)count; i++) {
-       c_keys[i] = [[keys objectAtIndex: i] UTF8String];
-       c_values[i] = [[[values objectAtIndex: i]description] UTF8String];
-     }
-    callback(instance,c_keys,count,c_values,count);
-    free(c_keys);
-    free(c_values);
+const char * getMetadataJson(NSDictionary* dictionary){
+    @try {
+        NSError *error = nil;
+        NSData *data = [NSJSONSerialization dataWithJSONObject:dictionary options:0 error:&error];
+        if (data) {
+            return strndup((const char *)data.bytes, data.length);
+        } else {
+            NSLog(@"%@", error);
+        }
+    } @catch (NSException *exception) {
+        NSLog(@"%@", exception);
+    }
+    return NULL;
+}
+
+NSDictionary * getDictionaryFromMetadataJason(const char * jsonString){
+    @try {
+        NSError *error = nil;
+        NSData *data = [NSData dataWithBytesNoCopy:(void *)jsonString length:strlen(jsonString) freeWhenDone:NO];
+        id metadata = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+        if ([metadata isKindOfClass:[NSDictionary class]]) {
+            return metadata;
+        }
+    } @catch (NSException *exception) {
+        NSLog(@"%@", exception);
+        return NULL;
```

Comment:
Pedantic: prefer using `nil` for Obj-C objects

```suggestion
        return nil;
```

Reply:
Done